earn/src/components/borrow/modal/tab/AddCollateralTab.tsx

```diff
@@ -0,0 +1,260 @@
+
+  const [isPending, setIsPending] = useState(false);
+
+  const { config: contractWriteConfig } = usePrepareContractWrite({
+    address: collateralToken.address,
+    abi: ERC20ABI,
+    functionName: 'transfer',
+    args: [marginAccount.address, collateralAmount.toFixed()],
+    enabled: !!collateralAmount && !!userBalance && collateralAmount.lte(userBalance),
+    chainId: activeChain.id,
+  });
+  const contractWriteConfigUpdatedRequest = useMemo(() => {
+    if (contractWriteConfig.request) {
+      return {
+        ...contractWriteConfig.request,
+        gasLimit: contractWriteConfig.request.gasLimit.mul(GAS_ESTIMATE_WIGGLE_ROOM).div(100),
+      };
+    }
+    return undefined;
+  }, [contractWriteConfig.request]);
+  const {
+    write: contractWrite,
+    data: contractData,
+    isSuccess: contractDidSucceed,
+    isLoading: contractIsLoading,
+  } = useContractWrite({
+    ...contractWriteConfig,
+    request: contractWriteConfigUpdatedRequest,
+  });
+
+  useEffect(() => {
+    if (contractDidSucceed && contractData) {
+      setPendingTxn(contractData);
+      setIsPending(false);
+      setIsOpen(false);
+    } else if (!contractIsLoading && !contractDidSucceed) {
+      setIsPending(false);
+    }
+  }, [contractDidSucceed, contractData, contractIsLoading, setPendingTxn, setIsOpen]);
+
+  let confirmButtonState = ConfirmButtonState.READY;
+
+  if (collateralAmount.gt(userBalance)) {
+    confirmButtonState = ConfirmButtonState.INSUFFICIENT_ASSET;
+  } else if (isPending) {
+    confirmButtonState = ConfirmButtonState.PENDING;
+  }
+
+  const confirmButton = getConfirmButton(confirmButtonState, collateralToken);
+
+  return (
+    <FilledStylizedButton
+      size='M'
+      fillWidth={true}
+      disabled={!confirmButton.enabled}
+      onClick={() => {
+        if (confirmButtonState === ConfirmButtonState.READY) {
+          setIsPending(true);
+          contractWrite?.();
+        }
+      }}
+    >
+      {confirmButton.text}
+    </FilledStylizedButton>
+  );
+}
+
+export type AddCollateralTabProps = {
+  marginAccount: MarginAccount;
+  isOpen: boolean;
+  setIsOpen: (open: boolean) => void;
+  setPendingTxn: (pendingTxn: SendTransactionResult | null) => void;
+};
+
+export function AddCollateralTab(props: AddCollateralTabProps) {
+  const { marginAccount, isOpen, setIsOpen, setPendingTxn } = props;
+  const { activeChain } = useContext(ChainContext);
+
+  const [collateralAmount, setCollateralAmount] = useState('');
+  const [collateralToken, setCollateralToken] = useState(marginAccount.token0);
+
+  const { address: userAddress } = useAccount();
+
+  const { refetch: refetchBalance, data: userBalance } = useBalance({
+    address: userAddress,
+    token: collateralToken.address,
+    chainId: activeChain.id,
+    watch: false,
+    enabled: isOpen,
+  });
+
+  useEffect(() => {
+    let interval: NodeJS.Timer | null = null;
+    interval = setInterval(() => refetchBalance(), 13_000);
```

Comment:
You don't even need the arrow function (`setInterval(refetchBalance, 13_000)` works) but this is fine